src/bits/bit_field_vec.rs

```diff
@@ -93,20 +93,102 @@ fn mask<W: Word>(bit_width: usize) -> W {
 }
 
 impl<W: Word> BitFieldVec<W, Vec<W>> {
-    /// Create a new zero-initialized vector of given bit width and length.
-    pub fn new(bit_width: usize, len: usize) -> Self {
-        // We need at least one word to handle the case of bit width zero.
+    /// Returns a new BitFieldVec filled with `word`
+    ///
+    /// # Arguments
+    /// * `bit_width` - The bit width of the values stored in the vector.
+    /// * `len` - The length of the vector.
+    /// * `word` - The value to fill the vector with.
+    ///
+    /// # Example
+    /// In the following example, we show how to create a new BitFieldVec
+    /// filled with a given word. In this case, we create a vector with length
+    /// 10 and bit width 4, filled with the value 0b11110000. Do note that the
+    /// number of bits in the word must be equal to the bit width of the vector.
```

Comment:
I'd use "value", rather than "word", as it's not a word unless the bit width is `usize`. And what you mean is that "the bit witdth must be sufficiently large to hold the given value" or something like that. As it is written now it looks like the bit of order bit_width - 1 should be set, as that's the only meaning I can guess for "number of bits".